File: Samples/JotFormTest/Classes/Controller/DataListViewController.m
//
//  DataListViewController.m
//  JotFormTest
//
//  Created by Interlogy, LLC on 1/6/14.
//  Copyright (c) 2014 Interlogy, LLC. All rights reserved.
//

#import "DataListViewController.h"
#import "SharedData.h"
#import "Common.h"

@interface DataListViewController () {
           DataListType listType;
}

@property (nonatomic,weak) IBOutlet UITableView *listTableView;
@property (nonatomic,strong) NSMutableArray *dataList;

@end

@implementation DataListViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
}

#pragma mark - user definition method

- (void)setList:(NSArray *)dataArray type:(DataListType)type {
    listType = type;
    
    self.dataList = [[NSMutableArray alloc] initWithArray:dataArray];
    [self.listTableView reloadData];
    
    switch (listType) {
        case DataListTypeFormList:
            self.title = @"Form list";
            break;
        case DataListTypeSubmissionList:
            self.title = @"Submission list";
            break;
        default:
            self.title = @"Report list";
            break;
    }
    
    self.title = @"Form list";
}

#pragma mark - UITableViewDataSource

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section {
    return 0.1f;
}

- (CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSInteger)section {
    return 0.1f;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return [self.dataList count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"SampleCell"];
    
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:@"SampleCell"];
    }
    
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    
    id object = [self.dataList objectAtIndex:indexPath.row];
    cell.detailTextLabel.text = [object objectForKey:@"id"];
    
    if (listType == DataListTypeFormList || listType == DataListTypeReportList) {
        cell.textLabel.text = [object objectForKey:@"title"];
    }
    
    cell.textLabel.textColor = [UIColor darkGrayColor];
    
    return cell;
}

#pragma mark - UITableViewDelegate

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 44.0f;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
}

@end

File: Samples/JotFormTest/Classes/Controller/DataListViewController.swift
//
//  DataListViewController.swift
//  JotFormTest
//
//  Created by Curtis Stilwell on 2/1/18.
//  Copyright © 2018 wang. All rights reserved.
//

import Foundation

class DataListViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
    
    @IBOutlet weak var listTableView: UITableView?
   
    var listType = DataListType(rawValue: 0)
    var dataList = [AnyHashable]()
    
    public func setList(_ dataArray:[AnyObject], type: DataListType) {
        listType = type
        
        if let array = dataArray as? [AnyHashable] {
            dataList = array
        }
        
        listTableView?.reloadData()
        
        switch listType {
        case .formList?:
            title = "Form list"
        case .submissionList?:
            title = "Submission list"
        default:
            title = "Report list"
        }
    }
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
        return 0.1
    }
    
    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
        return 0.1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataList.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell = UITableViewCell(style: .subtitle, reuseIdentifier: "SampleCell")
        
        cell.selectionStyle = .none
       
        if let object = dataList[indexPath.row] as? [AnyHashable : Any] {
            
            if let idString =  object["id"] as? String {
               cell.detailTextLabel?.text = idString
            }
            
            if listType == DataListType.formList || listType == DataListType.reportList {
                cell.textLabel?.text = object["title"] as? String
            }
        }
        
        cell.textLabel?.textColor = UIColor.darkGray

        return cell
    }
}
